src/plugins/wizard/public/application/utils/state_management/metadata_slice.ts

```diff
@@ -12,12 +12,14 @@ export interface MetadataState {
       // Validity for each section in the editor
       [key: string]: boolean;
     };
+    state: 'initial' | 'clean' | 'dirty';
```

Comment:
nit: Move this to a separate type property so that you can reference it in your action too